### ISSUE TITLE
Explore outlines logits processor for Structured Path Generation

### ISSUE THREAD

**Issue by jackmedda** [tail-unica/hopwise#101]
Generating the path such that it follows the KG structure, as done by the ConstrainedLogitsProcessor, could be implemented via outlines_core and outlines by using Generation Guide https://github.com/dottxt-ai/outlines/blob/main/outlines/fsm/guide.py.
It should probably provide several improvements, such as the fast Rust implementation
